FTBFS with Vala 0.42

Building this fails on Nix with Vala 0.42:

```
cmake flags: -GNinja -DCMAKE_BUILD_TYPE=Release -DCMAKE_SKIP_BUILD_RPATH=ON -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/lpgqc0y7bl4mcz47qfgxsvfsgq4f2j0n-spice-up-1.5.2/include -DCMAKE_INSTALL_LIBDIR=/nix/store/lpgqc0y7bl4mcz47qfgxsvfsgq4f2j0n-spice-up-1.5.2/lib -DCMAKE_INSTALL_NAME_DIR=/nix/store/lpgqc0y7bl4mcz47qfgxsvfsgq4f2j0n-spice-up-1.5.2/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_OSX_DEPLOYMENT_TARGET= -DCMAKE_OSX_SYSROOT= -DCMAKE_FIND_FRAMEWORK=last -DCMAKE_STRIP=/nix/store/h0lbngpv6ln56hjj59i6l77vxq25flbz-binutils-2.30/bin/strip -DCMAKE_RANLIB=/nix/store/h0lbngpv6ln56hjj59i6l77vxq25flbz-binutils-2.30/bin/ranlib -DCMAKE_AR=/nix/store/h0lbngpv6ln56hjj59i6l77vxq25flbz-binutils-2.30/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/lpgqc0y7bl4mcz47qfgxsvfsgq4f2j0n-spice-up-1.5.2
-- The C compiler identification is GNU 7.3.0
-- The CXX compiler identification is GNU 7.3.0
-- Check for working C compiler: /nix/store/iw94llkj05wgaz268mlzvgx8jkbi1ss0-gcc-wrapper-7.3.0/bin/gcc
-- Check for working C compiler: /nix/store/iw94llkj05wgaz268mlzvgx8jkbi1ss0-gcc-wrapper-7.3.0/bin/gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /nix/store/iw94llkj05wgaz268mlzvgx8jkbi1ss0-gcc-wrapper-7.3.0/bin/g++
-- Check for working CXX compiler: /nix/store/iw94llkj05wgaz268mlzvgx8jkbi1ss0-gcc-wrapper-7.3.0/bin/g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PkgConfig: /nix/store/dj3sdmwgfqbnn09bdf6spzw5p58ck0jq-pkg-config-0.29.2/bin/pkg-config (found version "0.29.2")
-- Checking for modules 'granite>=0.3;gee-0.8;gtk+-3.0>=3.9.10;json-glib-1.0;gudev-1.0;libevdev;libsoup-2.4'
--   Found granite, version 5.1.0
--   Found gee-0.8, version 0.20.1
--   Found gtk+-3.0, version 3.24.0
--   Found json-glib-1.0, version 1.4.2
--   Found gudev-1.0, version 232
--   Found libevdev, version 1.5.9
--   Found libsoup-2.4, version 2.64.0
Package libpcre was not found in the pkg-config search path.
Perhaps you should add the directory containing `libpcre.pc'
to the PKG_CONFIG_PATH environment variable
Package 'libpcre', required by 'glib-2.0', not found
Package libpcre was not found in the pkg-config search path.
Perhaps you should add the directory containing `libpcre.pc'
to the PKG_CONFIG_PATH environment variable
Package 'libpcre', required by 'glib-2.0', not found
Package libpcre was not found in the pkg-config search path.
Perhaps you should add the directory containing `libpcre.pc'
to the PKG_CONFIG_PATH environment variable
Package 'libpcre', required by 'glib-2.0', not found
Package libpcre was not found in the pkg-config search path.
Perhaps you should add the directory containing `libpcre.pc'
to the PKG_CONFIG_PATH environment variable
Package 'libpcre', required by 'glib-2.0', not found
-- Found Vala: /nix/store/99qldd1s09hg1924qma60gzipsda0kvv-vala-0.42.0/bin/valac
-- checking for a minimum Vala version of 0.26.0
--   found Vala, version 0.42.0
-- GSettings schemas will be installed locally.
-- GSettings shemas will be compiled.
-- GSettings schemas will be installed into /nix/store/lpgqc0y7bl4mcz47qfgxsvfsgq4f2j0n-spice-up-1.5.2/share/glib-2.0/schemas/
Package libpcre was not found in the pkg-config search path.
Perhaps you should add the directory containing `libpcre.pc'
to the PKG_CONFIG_PATH environment variable
Package 'libpcre', required by 'glib-2.0', not found
Package libpcre was not found in the pkg-config search path.
Perhaps you should add the directory containing `libpcre.pc'
to the PKG_CONFIG_PATH environment variable
Package 'libpcre', required by 'glib-2.0', not found
Package libpcre was not found in the pkg-config search path.
Perhaps you should add the directory containing `libpcre.pc'
to the PKG_CONFIG_PATH environment variable
Package 'libpcre', required by 'glib-2.0', not found
Package libpcre was not found in the pkg-config search path.
Perhaps you should add the directory containing `libpcre.pc'
to the PKG_CONFIG_PATH environment variable
Package 'libpcre', required by 'glib-2.0', not found
-- Compiling for Gtk 3.22
-- Adding tests for nix-build-user
-- Configuring done
-- Generating done
-- Build files have been written to: /build/source/build
building
build flags: -j1 -l4
ninja: warning: multiple rules generate src/Widgets/CanvasItems/Grabber.c. builds involving this target will not be correct; continuing anyway [-w dupbuild=warn]
ninja: warning: multiple rules generate src/Widgets/CanvasItems/CanvasItem.c. builds involving this target will not be correct; continuing anyway [-w dupbuild=warn]
ninja: warning: multiple rules generate src/Widgets/Canvas.c. builds involving this target will not be correct; continuing anyway [-w dupbuild=warn]
ninja: warning: multiple rules generate src/Widgets/ColorPicker/ColorButton.c. builds involving this target will not be correct; continuing anyway [-w dupbuild=warn]
ninja: warning: multiple rules generate src/Widgets/ColorPicker/ColorPicker.c. builds involving this target will not be correct; continuing anyway [-w dupbuild=warn]
ninja: warning: multiple rules generate src/Widgets/ColorPicker/Gradient.c. builds involving this target will not be correct; continuing anyway [-w dupbuild=warn]
ninja: warning: multiple rules generate src/Widgets/ColorPicker/GradientEditor.c. builds involving this target will not be correct; continuing anyway [-w dupbuild=warn]
ninja: warning: multiple rules generate src/Widgets/ColorPicker/PickerWindow.c. builds involving this target will not be correct; continuing anyway [-w dupbuild=warn]
ninja: warning: multiple rules generate src/Widgets/CanvasItems/ColorItem.c. builds involving this target will not be correct; continuing anyway [-w dupbuild=warn]
ninja: warning: multiple rules generate src/Widgets/CanvasItems/ImageItem.c. builds involving this target will not be correct; continuing anyway [-w dupbuild=warn]
ninja: warning: multiple rules generate src/Widgets/CanvasItems/TextItem.c. builds involving this target will not be correct; continuing anyway [-w dupbuild=warn]
[1/89] Running utility command for i18n
[2/89] Generating ../data/assets.gresource.c
[3/89] Generating Main.c, Application.c, Window.c, Services/Settings.c, Services/Slide.c, Services/SlideManager.c, Services/Fetcher.c, Services/FileManager.c, Services/HistoryManager.c, Services/ImageHandler.c, Services/Utils.c, Services/GamepadSlideController.c, Widgets/Canvas.c, Widgets/EntryCombo.c, Widgets/Headerbar.c, Widgets/SlideList.c, Widgets/DynamicToolbar.c, Widgets/Welcome.c, Widgets/SlideWidget.c, Widgets/PresenterView.c, Widgets/ColorPicker/ColorButton.c, Widgets/ColorPicker/ColorPicker.c, Widgets/ColorPicker/Gradient.c, Widgets/ColorPicker/GradientEditor.c, Widgets/ColorPicker/PickerWindow.c, Widgets/CanvasItems/CanvasItem.c, Widgets/CanvasItems/TextItem.c, Widgets/CanvasItems/ColorItem.c, Widgets/CanvasItems/ImageItem.c, Widgets/CanvasItems/Grabber.c, Widgets/Toolbars/CanvasBar.c, Widgets/Toolbars/CommonBar.c, Widgets/Toolbars/ImageBar.c, Widgets/Toolbars/ShapeBar.c, Widgets/Toolbars/TextBar.c, Widgets/Toolbars/Toolbar.c, Widgets/Library/LibraryItem.c, Widgets/Library/Library.c, Services/libgamepad/libgamepad.c, Services/libgamepad/gamepad-monitor.c, Services/libgamepad/gamepad.c, Services/libgamepad/helpers.c, Services/libgamepad/input-type.c, Services/libgamepad/raw-gamepad-interface.c, Services/libgamepad/raw-gamepad-monitor-interface.c, Services/libgamepad/standard-gamepad-axis.c, Services/libgamepad/standard-gamepad-button.c, Services/libgamepad/drivers/linux/guid-helpers.c, Services/libgamepad/drivers/linux/raw-gamepad.c, Services/libgamepad/drivers/linux/raw-gamepad-monitor.c
FAILED: src/Main.c src/Application.c src/Window.c src/Services/Settings.c src/Services/Slide.c src/Services/SlideManager.c src/Services/Fetcher.c src/Services/FileManager.c src/Services/HistoryManager.c src/Services/ImageHandler.c src/Services/Utils.c src/Services/GamepadSlideController.c src/Widgets/Canvas.c src/Widgets/EntryCombo.c src/Widgets/Headerbar.c src/Widgets/SlideList.c src/Widgets/DynamicToolbar.c src/Widgets/Welcome.c src/Widgets/SlideWidget.c src/Widgets/PresenterView.c src/Widgets/ColorPicker/ColorButton.c src/Widgets/ColorPicker/ColorPicker.c src/Widgets/ColorPicker/Gradient.c src/Widgets/ColorPicker/GradientEditor.c src/Widgets/ColorPicker/PickerWindow.c src/Widgets/CanvasItems/CanvasItem.c src/Widgets/CanvasItems/TextItem.c src/Widgets/CanvasItems/ColorItem.c src/Widgets/CanvasItems/ImageItem.c src/Widgets/CanvasItems/Grabber.c src/Widgets/Toolbars/CanvasBar.c src/Widgets/Toolbars/CommonBar.c src/Widgets/Toolbars/ImageBar.c src/Widgets/Toolbars/ShapeBar.c src/Widgets/Toolbars/TextBar.c src/Widgets/Toolbars/Toolbar.c src/Widgets/Library/LibraryItem.c src/Widgets/Library/Library.c src/Services/libgamepad/libgamepad.c src/Services/libgamepad/gamepad-monitor.c src/Services/libgamepad/gamepad.c src/Services/libgamepad/helpers.c src/Services/libgamepad/input-type.c src/Services/libgamepad/raw-gamepad-interface.c src/Services/libgamepad/raw-gamepad-monitor-interface.c src/Services/libgamepad/standard-gamepad-axis.c src/Services/libgamepad/standard-gamepad-button.c src/Services/libgamepad/drivers/linux/guid-helpers.c src/Services/libgamepad/drivers/linux/raw-gamepad.c src/Services/libgamepad/drivers/linux/raw-gamepad-monitor.c
cd /build/source/build/src && /nix/store/99qldd1s09hg1924qma60gzipsda0kvv-vala-0.42.0/bin/valac -C -b /build/source/src -d /build/source/build/src --pkg=granite --pkg=gee-0.8 --pkg=gtk+-3.0 --pkg=json-glib-1.0 --pkg=gudev-1.0 --pkg=libevdev --pkg=linux --pkg=libsoup-2.4 --define=GTK_3_22 -g --thread --target-glib=2.38 --vapidir=/build/source/vapi /build/source/src/Main.vala /build/source/src/Application.vala /build/source/src/Window.vala /build/source/src/Services/Settings.vala /build/source/src/Services/Slide.vala /build/source/src/Services/SlideManager.vala /build/source/src/Services/Fetcher.vala /build/source/src/Services/FileManager.vala /build/source/src/Services/HistoryManager.vala /build/source/src/Services/ImageHandler.vala /build/source/src/Services/Utils.vala /build/source/src/Services/GamepadSlideController.vala /build/source/src/Widgets/Canvas.vala /build/source/src/Widgets/EntryCombo.vala /build/source/src/Widgets/Headerbar.vala /build/source/src/Widgets/SlideList.vala /build/source/src/Widgets/DynamicToolbar.vala /build/source/src/Widgets/Welcome.vala /build/source/src/Widgets/SlideWidget.vala /build/source/src/Widgets/PresenterView.vala /build/source/src/Widgets/ColorPicker/ColorButton.vala /build/source/src/Widgets/ColorPicker/ColorPicker.vala /build/source/src/Widgets/ColorPicker/Gradient.vala /build/source/src/Widgets/ColorPicker/GradientEditor.vala /build/source/src/Widgets/ColorPicker/PickerWindow.vala /build/source/src/Widgets/CanvasItems/CanvasItem.vala /build/source/src/Widgets/CanvasItems/TextItem.vala /build/source/src/Widgets/CanvasItems/ColorItem.vala /build/source/src/Widgets/CanvasItems/ImageItem.vala /build/source/src/Widgets/CanvasItems/Grabber.vala /build/source/src/Widgets/Toolbars/CanvasBar.vala /build/source/src/Widgets/Toolbars/CommonBar.vala /build/source/src/Widgets/Toolbars/ImageBar.vala /build/source/src/Widgets/Toolbars/ShapeBar.vala /build/source/src/Widgets/Toolbars/TextBar.vala /build/source/src/Widgets/Toolbars/Toolbar.vala /build/source/src/Widgets/Library/LibraryItem.vala /build/source/src/Widgets/Library/Library.vala /build/source/src/Services/libgamepad/libgamepad.vala /build/source/src/Services/libgamepad/gamepad-monitor.vala /build/source/src/Services/libgamepad/gamepad.vala /build/source/src/Services/libgamepad/helpers.vala /build/source/src/Services/libgamepad/input-type.vala /build/source/src/Services/libgamepad/raw-gamepad-interface.vala /build/source/src/Services/libgamepad/raw-gamepad-monitor-interface.vala /build/source/src/Services/libgamepad/standard-gamepad-axis.vala /build/source/src/Services/libgamepad/standard-gamepad-button.vala /build/source/src/Services/libgamepad/drivers/linux/guid-helpers.vala /build/source/src/Services/libgamepad/drivers/linux/raw-gamepad.vala /build/source/src/Services/libgamepad/drivers/linux/raw-gamepad-monitor.vala
/build/source/src/Widgets/ColorPicker/ColorPicker.vala:373.5-450.6: warning: the modifier `static' is not applicable to constants
/build/source/src/Widgets/ColorPicker/ColorPicker.vala:452.5-484.6: warning: the modifier `static' is not applicable to constants
/build/source/src/Window.vala:25.5-25.29: error: Property `Spice.Window.is_fullscreen' with custom `get' accessor and/or `set' mutator cannot have `default' value
    public bool is_fullscreen {
    ^^^^^^^^^^^^^^^^^^^^^^^^^
/build/source/src/Services/Utils.vala:149.9-149.46: warning: Gdk.Screen.get_active_window has been deprecated since 3.22
/build/source/src/Widgets/SlideList.vala:165.9-165.25: warning: Spice.SlideList.SlideListRow.style hides inherited method `Gtk.Widget.style'. Use the `new' keyword if hiding was intentional
        public void style () {
        ^^^^^^^^^^^^^^^^^
/build/source/src/Services/Slide.vala:34.5-34.23: error: Property `Spice.Slide.visible' with custom `get' accessor and/or `set' mutator cannot have `default' value
    public bool visible {
    ^^^^^^^^^^^^^^^^^^^
/build/source/src/Widgets/PresenterView.vala:203.5-203.22: warning: Spice.PresenterWindow.style hides inherited method `Gtk.Widget.style'. Use the `new' keyword if hiding was intentional
    private void style () {
    ^^^^^^^^^^^^^^^^^^
/build/source/src/Widgets/PresenterView.vala:35.5-35.25: error: Property `Spice.PresenterWindow.font_size' with custom `get' accessor and/or `set' mutator cannot have `default' value
    private int font_size {
    ^^^^^^^^^^^^^^^^^^^^^
/build/source/src/Widgets/Canvas.vala:32.5-32.31: error: Property `Spice.Canvas.current_ratio' with custom `get' accessor and/or `set' mutator cannot have `default' value
    public double current_ratio {
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^
/build/source/src/Services/FileManager.vala:108.9-108.73: warning: unhandled error `GLib.FileError'
        GLib.FileUtils.set_data (result.get_path (), formatted_data.data);
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/build/source/src/Services/FileManager.vala:132.44-132.55: warning: unhandled error `GLib.Error'
            var dis = new DataInputStream (file.read ());
                                           ^^^^^^^^^^^^
/build/source/src/Services/FileManager.vala:135.22-135.55: warning: unhandled error `GLib.IOError'
            header = dis.read_upto ("\0", -1, out size);
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/build/source/src/Services/FileManager.vala:140.44-140.55: warning: unhandled error `GLib.Error'
            var dis = new DataInputStream (file.read ());
                                           ^^^^^^^^^^^^
/build/source/src/Services/FileManager.vala:143.22-143.55: warning: unhandled error `GLib.IOError'
            footer = dis.read_upto ("\0", -1, out size);
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/build/source/src/Services/Utils.vala:170.17-170.33: warning: Gdk.Screen.height has been deprecated since 3.22
/build/source/src/Services/Utils.vala:171.17-171.32: warning: Gdk.Screen.width has been deprecated since 3.22
/build/source/src/Window.vala:435.29-435.49: warning: Gdk.Screen.get_n_monitors has been deprecated since 3.22
/build/source/src/Window.vala:442.35-442.60: warning: Gdk.Screen.get_primary_monitor has been deprecated since 3.22
/build/source/src/Window.vala:445.13-445.39: warning: Gdk.Screen.get_monitor_geometry has been deprecated since 3.22
/build/source/src/Widgets/Library/LibraryItem.vala:194.40-194.51: warning: unhandled error `GLib.Error'
        var dis = new DataInputStream (file.read ());
                                       ^^^^^^^^^^^^
/build/source/src/Widgets/Library/LibraryItem.vala:197.16-197.49: warning: unhandled error `GLib.IOError'
        data = dis.read_upto ("\0", -1, out size);
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Compilation failed: 4 error(s), 17 warning(s)
ninja: build stopped: subcommand failed.
builder for '/nix/store/4cpzcy57474kq4p4ml85y3hrkzqiarnl-spice-up-1.5.2.drv' failed with exit code 1
```

--- a/src/Services/Slide.vala
+++ b/src/Services/Slide.vala
@@ -40,8 +40,6 @@ public class Spice.Slide : Object {
             canvas.visible = value;
             visible_changed (value);
         }
-
-        default = true;
     }
 
     public Slide (Json.Object? save_data = null) {

--- a/src/Widgets/Canvas.vala
+++ b/src/Widgets/Canvas.vala
@@ -28,7 +28,7 @@ public class Spice.Canvas : Gtk.Overlay {
     public signal void next_slide ();
     public signal void previous_slide ();
 
-    private double _current_ratio;
+    private double _current_ratio = 1.0f;
     public double current_ratio {
         get {
             return _current_ratio;
@@ -40,8 +40,6 @@ public class Spice.Canvas : Gtk.Overlay {
                 ratio_changed (current_ratio);
             }
         }
-
-        default = 1.0f;
     }
 
     public double current_allocated_width = 0;

--- a/src/Widgets/PresenterView.vala
+++ b/src/Widgets/PresenterView.vala
@@ -31,9 +31,8 @@ public class Spice.PresenterWindow : Gtk.Window {
 
     Clock clock;
 
-    private int _font_size;
+    private int _font_size = 12;
     private int font_size {
-        default = 12;
         get {
             return _font_size;
         }

--- a/src/Window.vala
+++ b/src/Window.vala
@@ -32,11 +32,11 @@ public class Spice.Window : Gtk.ApplicationWindow {
             } else {
                 aspect_frame.margin = 24;
             }
-        } default = false;
+        }
     }
 
     private bool notification_shown = false;
-    private bool is_full_;
+    private bool is_full_ = false;
 
     private Spice.Headerbar headerbar;
     private Spice.SlideManager slide_manager;

